### PR TITLE
Add code snippet to rerun Axelrod's second tournament to readme

### PR DESCRIPTION
**Pull request by marcharper**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -18,15 +18,36 @@ Installation
...
+   print(len(players), "players")
 
-    >>> import axelrod_fortran as axlf
-    >>> import axelrod as axl
-    >>> p1 = axlf.Player('k31r')
-    >>> p2 = axlf.Player('k33r')
-    >>> match = axl.Match((p1, p2), turns=5)
-    >>> match.play()
-    [(C, C), (C, C), (C, D), (C, D), (C, C)]
+   tournament = axl.Tournament(players, repetitions=100)
```
Comment (drvinceknight) [inline README.rst]: Axelrod's second tournament was `prob_end=.00346` https://github.com/Axelrod-Python/axelrod-fortran/issues/23

Also for the purposes of this readme, let's set `repetitions=5`.

File: README.rst [hunk 2]
```diff
@@ -18,15 +18,36 @@ Installation
...
+   results = tournament.play(processes=4)
```
Comment (drvinceknight) [inline README.rst]: Let's keep this as `tournament.play()` (for the README).

File: README.rst [hunk 3]
```diff
@@ -18,15 +18,36 @@ Installation
...
+   >>> p2 = axlf.Player('k33r')
+   >>> match = axl.Match((p1, p2), turns=5)
+   >>> match.play()
+   [(C, C), (C, C), (C, D), (C, D), (C, C)]
+
+Reproducing Axelrod's second tournament:
+
+.. code-block:: python
+
+   import axelrod as axl
```
Comment (drvinceknight) [inline README.rst]: Can we write all code with python `>>>` so that we can run doctests to catch errors.

Comment (drvinceknight) [inline README.rst]: (`python -m doctest README.rst`)

File: README.rst (hunk 1 above)
Comment (marcharper) [inline README.rst]: Set reps to 5, technically the tournaments were of fixed length so I left it as is for now.

File: README.rst (hunk 3 above)
Comment (marcharper) [inline README.rst]: Done.

File: README.rst (hunk 2 above)
Comment (marcharper) [inline README.rst]: Done.

File: README.rst (hunk 1 above)
Comment (drvinceknight) [inline README.rst]: The players did not know the tournament length so I don't think it's as clear as "fixed length". If we're going to write `Axelrod's second tournament` then I feel that `prob_end` is "less wrong".

File: README.rst [hunk 4]
```diff
@@ -18,15 +18,33 @@ Installation
...
+   >>> match = axl.Match((p1, p2), turns=5)
+   >>> match.play()
+   [(C, C), (C, C), (C, D), (C, D), (C, C)]
+
+Running an instance of Axelrod's second tournament:
+
+.. code-block:: python
+
+   >>> import axelrod as axl
+   >>> from axelrod_fortran.strategies import characteristics
```
Comment (drvinceknight) [inline README.rst]: to match up with `axelrod` (and the lines of code written directly above), can we remove these two imports and just `import axelrod_fortran ax axlf` and then we can access:

```
axlf.charactersitics
axlf.Player
```

File: README.rst (hunk 1 above)
Comment (drvinceknight) [inline README.rst]: Actually, feel free to leave it for now, I'm going to open a PR to axelrod that will make it possible to pass what a player knows in a tournament (we can currently do that in a match). Then we can just fix this.

File: README.rst [hunk 5]
```diff
@@ -18,15 +18,32 @@ Installation
...
+
+Running an instance of Axelrod's second tournament:
+
+.. code-block:: python
+
+   >>> import axelrod_fortran as axlf
+   >>> import axelrod as axl
+   >>> players = [axlf.Player(name) for name in axlf.characteristics.keys()
+   >>>            if axlf.characteristics[name]["original_rank"] is not None]
+   >>> print(len(players), "players")
```
Comment (drvinceknight) [inline README.rst]: Woops, this would fail the doctes as no output is shown. Want to write the output or just delete this line?

File: README.rst [hunk 6]
```diff
@@ -18,15 +18,33 @@ Installation
...
+   >>> tournament = axl.Tournament(players, repetitions=1, turns=200)
+   >>> results = tournament.play()
+   >>> results.write_summary('summary.csv')
+   >>> plot = axl.Plot(results)
+   >>> plot.save_all_plots("second_tournament")
+   62 players
```
Comment (drvinceknight) [inline README.rst]: Sorry, wrong spot. It needs to be just after the print statement:

```python
>>> print(len(players), "players")
62 players
>>> tournament = axl.Tournament(players, repetitions=1, turns=200)
```

File: README.rst [hunk 7]
```diff
@@ -18,15 +18,34 @@ Installation
...
+   [(C, C), (C, C), (C, D), (C, D), (C, C)]
+
+Running an instance of Axelrod's second tournament:
+
+.. code-block:: python
+
+   >>> import axelrod_fortran as axlf
+   >>> import axelrod as axl
+   >>> players = [axlf.Player(name) for name in axlf.characteristics.keys()
+   >>>            if axlf.characteristics[name]["original_rank"] is not None]
```
Comment (meatballs) [inline README.rst]: I think the readme should use `all_strategies` and we should change its comprehension to be the one shown here

Comment (drvinceknight) [inline README.rst]: I'm not sure I agree. `all_strategies` is the list of all the strategies in `TourExec` (some that were are were apparently not in the tournament). We could create another list for `tournament_strategies` perhaps?

Comment (marcharper) [inline README.rst]: Ok but it's not an instance of the second tournament if it has the additional strategies.

Comment (drvinceknight) [inline README.rst]: (But this can be merged right, it's the "correct way to presently do this", once we get a new list or whatever in we can tweak the readme?)

Comment (meatballs) [inline README.rst]: I'm saying we chould change the definition of all_strategies to be the comprehension that @marcharper has used here. i.e. all-strategies should be the list of strategies that participated in the second tournament.

Comment (drvinceknight) [inline README.rst]: > all-strategies should be the list of strategies that participated in the second tournament.

`all_strategies` in `axelrod` points at the entire list of strategies (including the cheaters etc).

I think that would be nice to keep here too (so there still is a list of **all** the strategies) but then we can create a second list that is @marcharper's comprehension that would be something like `axlf.tournament_strategies`?

File: src/axelrod_fortran/player.py [hunk 8]
```diff
@@ -49,7 +48,7 @@ def original_name(self, value):
         if value in characteristics:
             self.__original_name = value
         else:
-            raise ValueError(f'{value} is not a valid Fortran function')
+            raise ValueError('{} is not a valid Fortran function'.format(value))
```
Comment (meatballs) [inline src/axelrod_fortran/player.py]: Why this change?

Comment (meatballs) [inline src/axelrod_fortran/player.py]: At the moment, the library is 3.6 only, so the 'f' strings are fine.
